packages/flutter/lib/src/widgets/app.dart

```diff
@@ -1660,20 +1672,36 @@ class _WidgetsAppState extends State<WidgetsApp> with WidgetsBindingObserver {
   }
 }
 
-/// Builds [MediaQuery] from `window` by listening to [WidgetsBinding].
+/// Provides a [MediaQuery] which is built and updated using the latest `window`
+/// values.
+///
+/// Receives `window` updates by listening to [WidgetsBinding].
 ///
-/// It is performed in a standalone widget to rebuild **only** [MediaQuery] and
-/// its dependents when `window` changes, instead of rebuilding the entire widget tree.
-class _MediaQueryFromWindow extends StatefulWidget {
-  const _MediaQueryFromWindow({Key? key, required this.child}) : super(key: key);
+/// The standalone widget ensures that it rebuilds **only** [MediaQuery] and
+/// its dependents when `window` changes, instead of rebuilding the entire
+/// widget tree.
+///
+/// It is used by [WidgetsApp] if no other [MediaQuery] is available above it.
+///
+/// See also:
+///
+///  * [MediaQuery], which establishes a subtree in which media queries resolve
+///    to a [MediaQueryData].
+class MediaQueryFromWindow extends StatefulWidget {
+  /// Creates a [MediaQueryFromWindow] that provides a [MediaQuery] to its
+  /// descendants using the `window` to keep [MediaQueryData] up to date.
+  ///
+  /// The [child] must not be null.
+  const MediaQueryFromWindow({Key? key, required this.child}) : super(key: key);
 
+  /// {@macro flutter.widgets.ProxyWidget.child}
   final Widget child;
 
   @override
-  _MediaQueryFromWindowsState createState() => _MediaQueryFromWindowsState();
+  _MediaQueryFromWindowState createState() => _MediaQueryFromWindowState();
```

Comment:
```suggestion
  State<MediaQueryFromWindow> createState() => _MediaQueryFromWindowState();
```
`_MediaQueryFromWindowState` is private and shouldn't be exposed in public API (there's a PR out to fix this elsewhere in our code base)